src/components/main-view/main-view.jsx

```diff
@@ -82,64 +85,65 @@ class MainView extends React.Component {
     // console.log('should display movies now after successful registration but it isn\'t ');
 
+    // if (!user) return <LoginView onLoggedIn={user => this.onLoggedIn(user)} />;
 
 
-    /* Before the movies have been loaded */
-    if (movies.length === 0) return <div className="main-view" />;
 
     return (
 
       <Router>
 
-        <Navbar className="navbar"
-          collapseOnSelect
-          expand="lg"
-          variant="dark"
-          sticky="top">
-          <Container>
-            <Navbar.Brand href="#home">
-              <img
-                alt=""
-                src={Logo}
-                height="100"
-                className="d-inline-block align-top"
-              />{''}
-            </Navbar.Brand>
-            <Navbar.Toggle aria-controls="responsive-navbar-nav" />
-            <Navbar.Collapse id="responsive-navbar-nav">
-              <Nav className="nav-link">
-                <Nav.Link href="#home" >Home</Nav.Link>
-                <Nav.Link href="#features">Features</Nav.Link>
-                <Nav.Link href="#pricing">Pricing</Nav.Link>
-                <Nav.Link onClick={() => { this.onLoggedOut() }}>Logout</Nav.Link>
-              </Nav>
-            </Navbar.Collapse>
-          </Container>
-        </Navbar>
+        <NavbarView onLoggedOut={() => this.onLoggedOut()} />
+        <Row>
+          {user && <Link to={`/users/${username}`} >logged in as {username}</Link>}
+        </Row>
 
         <Row className="main-view justify-content-md-center">
-          <Route exact path="/" render={() => {
-            if (!user) return <LoginView onLoggedIn={user => this.onLoggedIn(user)} />;
-
+          <Route exact path="/" render={({ history }) => {
+            // CODE BELOW TO BE ACTIV AFTER ALL VIEWS WORKING
+            if (!user) return <Col>
+              <LoginView history={history} onLoggedIn={user => this.onLoggedIn(user)} />
+            </Col>
+            if (movies.length === 0) return <div className="main-view" />;
             console.log("rootpath");
             return movies.map(m => (
-              <Col md={3}>
-                <MovieCard key={m._id} movie={m} />
+              <Col md={3} key={m._id}>
+                <MovieCard movie={m} />
+
               </Col>
             ))
           }} />
 
-          <Route exact path="/register" render={() => {
-            return <RegistrationView />
+          <Route exact path="/register" render={({ history }) => {
+            if (user) return <Redirect to="/" />
+            return <RegistrationView history={history} onRegistration={(user) => this.onRegistration(user)} />
           }} />
 
+          <Route exact path={`/users/:username`} render={({ history }) => {
+            if (!user) return <Redirect to="/" />
+            return <ProfileView user={user} onBackClick={() => history.goBack()} />
+
+          }} />
+
+          <Route exact path={`/profile_update`} render={({ history }) => {
```

Comment:
If the course is not limiting us to what we can use here as the URL. Let us do something similar to the user profile view route only this time we should add `/profile-update` so the final URL will be `/users/:username/profile-update`. Note the `-` instead of `_`.

Reply:
I think the course brief was update user by the username (I think they meant not by _id ) so any URL PUT request containing username should be fine.

Reply:
Just thinking....as we are sending PUT request with the handleUpdateUser rather than POST to the /users/:username wouldn't that be enough to update a user credentials or do we need the /users/:username/profile-update path to not confuse the Route?

Reply:
`/users/:username/profile-update` is for react to show the right user interface. i.e. a form allowing a user to update user profile information. Thats all. It has nothing to do with which POST or PUT request is made to the server.

Once we show the right user interface with a button to `update`, then onClick, we can call the function to make an API call to update user details.

Reply:
Yes, got confused and carried away...
I changed the link to <Link to={`:username/profile-update`} >
                                <Button>Update Profile</Button>
                            </Link>
now it's showing up finally